nette/notejam/app/forms/Note/DeleteNoteFormFactory.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace App\Forms\Note;
+
+use App\Model\NoteManager;
+use Nette\Application\UI\Form;
+use Nette\Utils\ArrayHash;
+
+
+class DeleteNoteFormFactory
+{
+
+	/** @var NoteManager */
+	private $noteManager;
+
+	/**
+	 * EditPadFormFactory constructor.
+	 * @param NoteManager $noteManager
+	 */
+	public function __construct(NoteManager $noteManager)
+	{
+		$this->noteManager = $noteManager;
+	}
+
+	/**
+	 * @param int $id
+	 * @return Form
+	 */
+	public function create($id)
+	{
+		$form = new Form;
+		$form->addProtection(); // Adds CSRF protection
+
+		$form->addSubmit('submit', 'Yes, I want to delete this note');
+
+		$form->onSuccess[] = [$this, 'formSucceeded'];
+		return $form;
+	}
+
+	/**
+	 * @param Form      $form
+	 * @param ArrayHash $values
+	 */
+	public function formSucceeded(Form $form, $values)
+	{
+		if (!$this->noteManager->delete($form->getPresenter()->getParameter('id'))) {
```

Comment:
My point was that you're duplicating value of the ID.. You can pass it from the presenter even in the submit request. Hiddens are fine and might be simpler in this case, but generaly I like to not duplicate the data in the request, since I might be able to send that request to nonsense URL but having different value in hidden will make it work. Which doesn't make sense much.